Tidy docs for ServiceLoadBalancerClass and ServiceLBNodePortControl features (#31599)
kubernetes/website: content/en/docs/concepts/services-networking/service.md

#### Disabling load balancer NodePort allocation {#load-balancer-nodeport-allocation}

{{< feature-state for_k8s_version="v1.22" state="beta" >}}

> Starting in v1.20, you can optionally disable node port allocation for a Service Type=LoadBalancer by setting

Reviewer: ```suggestion
You can optionally disable node port allocation for a Service Type=LoadBalancer by setting
```
(this accompanies the other suggestion)

the field `spec.allocateLoadBalancerNodePorts` to `false`. This should only be used for load balancer implementations
that route traffic directly to pods as opposed to using node ports. By default, `spec.allocateLoadBalancerNodePorts`
is `true` and type LoadBalancer Services will continue to allocate node ports. If `spec.allocateLoadBalancerNodePorts`

> is set to `false` on an existing Service with allocated node ports, those node ports will NOT be de-allocated automatically.

Reviewer: When features graduate to beta, and I'm reviewing, I also look for and fix any style nits. For alpha features we're a bit less strict.
```suggestion
is set to `false` on an existing Service with allocated node ports, those node ports will **not** be de-allocated automatically.
```

You must explicitly remove the `nodePorts` entry in every Service port to de-allocate those node ports.

> You must enable the `ServiceLBNodePortControl` feature gate to use this field in v1.20 and v1.21,
> and the feature gate is enabled by default from v1.22 onwards.

Reviewer: The approach I recommend is:
```suggestion
Your cluster must have the `ServiceLBNodePortControl`
[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
enabled to use this field.
For Kubernetes v{{< skew currentVersion >}}, this feature gate is enabled by default,
and you can use the `spec.allocateLoadBalancerNodePorts` field. For clusters running
other versions of Kubernetes, check the documentation for that release.
```

That should stay current past graduation, and also encourages people to read the right docs if their cluster
isn't the very latest minor release. For example, we might change some detail between now and graduation to stable.
Also, mentions of feature gates should link to the feature gates page.

Author: That makes sense. I wasn't sure if there was a best-practice here, I just copied the same sentence from the following feature gate text.

Author: Since I copied the first-cut wording from the following feature gate, I copied your improvements back down into it.
